### PR TITLE
feature: Add more PII types to Column and Datum detectors 

### PR DESCRIPTION
**Pull request by nicolepng**

### CODE REVIEW

File: piicatcher/__init__.py [hunk 1]
```diff
@@ -59,6 +59,18 @@ class SSN(PiiType):
     pass
 
 
+class ZipCode(PiiType, type="zip_code"):  # type: ignore
```
Comment (jhecking) [inline piicatcher/__init__.py:62]: What's the purpose of the `type` parameter? Doesn't look like it's getting used.

File: piicatcher/scanner.py [hunk 2]
```diff
@@ -51,18 +53,23 @@ class ColumnNameRegexDetector(MetadataDetector):
...
+            re.IGNORECASE,
+        ),
+        ZipCode: re.compile(
+            "^.*(zipcode|zip_code|postal|postal_code|zip).*$",
             re.IGNORECASE,
         ),
         UserName: re.compile("^.*user(id|name|).*$", re.IGNORECASE),
         Password: re.compile("^.*pass.*$", re.IGNORECASE),
-        SSN: re.compile("^.*(ssn|social).*$", re.IGNORECASE),
+        SSN: re.compile("^.*(ssn|social_number).*$", re.IGNORECASE),
```
Comment (jhecking) [inline piicatcher/scanner.py:71]:
```suggestion
        SSN: re.compile("^.*(ssn|social_number|social_security).*$", re.IGNORECASE),
```

To cover column names like "social_security_number" or "social_security_no".

File: pyproject.toml [hunk 3]
```diff
@@ -28,6 +28,7 @@ pyyaml = "*"
 click = "*"
 python-json-logger = "^2.0.2"
 commonregex = "1.5.2"
```
Comment (jhecking) [inline pyproject.toml:30]: Can we drop the commonregex package or is it also still required?

File: piicatcher/__init__.py (hunk 1 above)
Comment (nicolepng) [reply piicatcher/__init__.py:62]: PiiType class requires a type to be detected. Without specifying the type, when dbcat tries to `data.get("type")`, typer will return None for ZipCode.

File: pyproject.toml (hunk 3 above)
Comment (nicolepng) [reply pyproject.toml]: removed!

**Review by jhecking — APPROVED** [review]
:shipit:

**Pull request merged**